### ISSUE TITLE
SolverBugException: Reached invalid decision id 0 with --prefer-lowest

### ISSUE THREAD

**Issue by sanmai** [composer/composer#9302]
My `composer.json`:

```json
{
    "require": {
        "ergebnis/composer-normalize": "^2.8.2",
        "friendsofphp/php-cs-fixer": "^2.16.4",
        "phan/phan": "^3.2.3",
        "php-coveralls/php-coveralls": "^2.4.1",
        "phpstan/phpstan": "^0.12.50",
        "phpunit/phpunit": "^9.4.1",
        "vimeo/psalm": "^3.17.2"
    }
}
```

Output of `composer diagnose`:

```
Checking composer.json: WARNING
No license specified, it is recommended to do so. For closed-source software you may use "proprietary" as license.
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com oauth access: OK
Checking disk free space: OK
Checking pubkeys: 
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK
Checking composer version: OK
Composer version: 2.0.0-RC2
PHP version: 7.4.11
PHP binary path: /usr/bin/php7.4
OpenSSL version: OpenSSL 1.1.1g  21 Apr 2020
cURL version: 7.72.0 libz 1.2.11 ssl OpenSSL/1.1.1g
zip extension: OK
```

When I run this command:

```
composer update --prefer-lowest --no-plugins
```

I get the following output:

```
Loading composer repositories with package information
Updating dependencies

                                                                                                                                                       
  [Composer\DependencyResolver\SolverBugException]                                                                                                     
  Reached invalid decision id 0 while looking through (108) for a literal present in the analyzed rule (-2850|3563|3564|3565).                         
  This exception was most likely caused by a bug in Composer.                                                                                          
  Please report the command you ran, the exact error you received, and your composer.json on https://github.com/composer/composer/issues - thank you!  
                                                                                                                                                       
```

And I expected this to happen:

Installation proceeds as usual.

#### Workaround

When I make this configuration less-specific like so:
```diff
-"phpstan/phpstan": "^0.12.50",
+"phpstan/phpstan": "^0.12",
```
Installation proceeds as it should. For the record, PHPStan has this [0.12.50](https://github.com/phpstan/phpstan/releases/tag/0.12.50) version.

---
Might be duplicate of / related to #9294

**Comment by Ocramius** [comment]
Similar issue on https://github.com/Ocramius/PackageVersions/pull/161/checks?check_run_id=1287019624

In my case, I'll drop the `--prefer-lowest` runs, as I regularly bump all dependencies to their latest versions anyway.

**Comment by naderman** [comment]
Looks like this is fixed by https://github.com/composer/composer/pull/9309 as well.

**Comment by Seldaek** [comment]
Confirmed #9309 fixed it, it now locks phpstan 0.12.50 correctly.

**Comment by Ocramius** [comment]
Excellent! :+1:
